<launch>
    <!-- ======================================================================================= -->
    <!--                                 CONFIGURATION ARGUMENTS                                 -->
    <!-- ======================================================================================= -->
    <arg name="mode" default="simple" />
    <arg name="trial_number" default="1" />
    <arg name="base_data_path" default="$(find fake_tiago_pkg)/data_basic" />
    <arg name="trial_data_path" default="$(find fake_tiago_pkg)/data_basic/trial_$(arg trial_number)" />

    <!-- ======================================================================================= -->
    <!--                             WORLD & VISION SIMULATION                                   -->
    <!-- ======================================================================================= -->
    <node name="fake_ar_publisher_node" pkg="fake_tiago_pkg" type="fake_ar_publisher.py" output="screen" />
    <node name="robot_vision_bridge_node" pkg="fake_tiago_pkg" type="robot_vision_bridge.py" output="screen" />
    <!-- <node name="vision_to_dnf_node" pkg="fake_tiago_pkg" type="vision_to_dnf_node.py" output="screen" /> -->
   <node name="vision_to_dnf_aggregator_node" pkg="fake_tiago_pkg" type="vision_to_dnf_aggregator.py" output="screen" />
    <!-- ======================================================================================= -->
    <!--                                   VOICE BRIDGE                                           -->
    <!-- ======================================================================================= -->
    <!-- <node name="voice_to_dnf_bridge" pkg="fake_tiago_pkg" type="voice_to_dnf_bridge.py" output="screen" /> -->
    <node name="voice_command_parser_node" pkg="fake_tiago_pkg" type="voice_command_parser.py" output="screen" />

    <!-- ======================================================================================= -->
    <!--                                  DNF RECALL NODE                                        -->
    <!-- ======================================================================================= -->
    <node name="dnf_recall_node" pkg="fake_tiago_pkg" type="dnf_model_recall_simple_node.py" output="screen">
        <param name="trial_number" value="$(arg trial_number)" />
        <param name="base_data_path" value="$(arg base_data_path)" />
        <param name="trial_data_path" value="$(arg trial_data_path)" />
        <param name="sequence_memory_path" value="$(arg base_data_path)/u_sm.npy" />
        <param name="duration_field_path"  value="$(arg base_data_path)/u_d.npy" />
    </node>

    <!-- ======================================================================================= -->
    <!--                                 EXECUTIVE NODES                                         -->
    <!-- ======================================================================================= -->
    <group if="$(eval arg('mode') == 'simple')">
        <node name="simulated_task_executive" pkg="fake_tiago_pkg" type="simulated_task_executive.py" output="screen" />
    </group>

    <group if="$(eval arg('mode') == 'tiago')">
        <node name="tiago_task_executive" pkg="fake_tiago_pkg" type="tiago_task_executive.py" output="screen" />
        <node name="fake_robot_controller" pkg="fake_tiago_pkg" type="fake_robot_controller.py" output="screen" />
    </group>
</launch>
